### ISSUE TITLE
JavaMnistClassification error

### ISSUE THREAD

**Issue by jdh3577** [deeplearning4j/dl4j-spark-ml-examples#15]
I ran the code of JavaMnistClassification connecting 4 nodes.

Layer configuration is following:
`MultiLayerConfiguration.Builder builder = new NeuralNetConfiguration.Builder()
                .seed(seed)
                .batchSize(batchSize)
                .iterations(iterations)
                .constrainGradientToUnitNorm(true)
                .optimizationAlgo(OptimizationAlgorithm.LINE_GRADIENT_DESCENT)
                .list(3)
                .layer(0, new ConvolutionLayer.Builder(10, 10)
                        .nIn(nChannels)
                        .nOut(6)
                        .weightInit(WeightInit.XAVIER)
                        .activation("relu")
                        .build())
                .layer(1, new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX, new int[] {2,2})
                        .build())
                .layer(2, new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                        .nIn(150)
                        .nOut(outputNum)
                        .weightInit(WeightInit.XAVIER)
                        .activation("softmax")
                        .build())
                .backprop(true).pretrain(false);
        new ConvolutionLayerSetup(builder,28,28,1);
`

The error is following:

`15/09/19 00:47:06 ERROR Executor: Exception in task 9.0 in stage 2.0 (TID 76)
java.lang.IllegalArgumentException: NDArrayIndex is out of range. Beginning index:4 must be less than its size:4
	at org.nd4j.linalg.indexing.NDArrayIndex.validate(NDArrayIndex.java:322)
	at org.nd4j.linalg.indexing.NDArrayIndex.resolve(NDArrayIndex.java:305)
	at org.nd4j.linalg.indexing.ShapeOffsetResolution.exec(ShapeOffsetResolution.java:46)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.get(BaseNDArray.java:3592)
	at org.deeplearning4j.nn.layers.convolution.subsampling.SubsamplingLayer.backpropGradient(SubsamplingLayer.java:99)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.backprop(MultiLayerNetwork.java:1105)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.computeGradientAndScore(MultiLayerNetwork.java:1438)
	at org.deeplearning4j.optimize.solvers.BaseOptimizer.gradientAndScore(BaseOptimizer.java:117)
	at org.deeplearning4j.optimize.solvers.BaseOptimizer.optimize(BaseOptimizer.java:135)
	at org.deeplearning4j.optimize.Solver.optimize(Solver.java:52)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fit(MultiLayerNetwork.java:1233)
	at org.deeplearning4j.spark.ml.classification.NeuralNetworkClassification$$anonfun$1.apply(MultiLayerNetworkClassification.scala:127)
	at org.deeplearning4j.spark.ml.classification.NeuralNetworkClassification$$anonfun$1.apply(MultiLayerNetworkClassification.scala:113)
	at org.deeplearning4j.spark.ml.nn.ParameterAveragingTrainingStrategy$$anonfun$train$1$$anonfun$apply$mcVI$sp$1.apply(TrainingStrategy.scala:89)
	at org.deeplearning4j.spark.ml.nn.ParameterAveragingTrainingStrategy$$anonfun$train$1$$anonfun$apply$mcVI$sp$1.apply(TrainingStrategy.scala:81)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:878)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:878)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:1765)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:1765)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:63)
	at org.apache.spark.scheduler.Task.run(Task.scala:70)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:213)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
`

**Comment by jdh3577** [comment]
my pom is 
`
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>org.deeplearning4j</groupId>
  <artifactId>dl4j-spark-ml-examples</artifactId>
  <version>1.0-SNAPSHOT</version>
  <packaging>jar</packaging>

  <name>spark-ml-examples</name>
  <url>http://maven.apache.org</url>

  <properties>
    <nd4j.version>0.4-rc3.1</nd4j.version>
    <dl4j.version>0.4-rc3.1</dl4j.version>
    <spark.version>1.4.0</spark.version>
  </properties>

    <repositories>
        <repository>
            <id>sonatype-nexus-snapshots</id>
            <name>Sonatype Nexus snapshot repository</name>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
        </repository>
    </repositories>

  <distributionManagement>
    <snapshotRepository>
      <id>sonatype-nexus-snapshots</id>
      <name>Sonatype Nexus snapshot repository</name>
      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </snapshotRepository>
    <repository>
      <id>nexus-releases</id>
      <name>Nexus Release Repository</name>
      <url>http://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
    </repository>
  </distributionManagement>

  <build>

    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-compiler-plugin</artifactId>
        <configuration>
          <source>1.7</source>
          <target>1.7</target>
        </configuration>
      </plugin>
    </plugins>
  </build>

  <dependencies>
    <dependency>
      <groupId>org.apache.spark</groupId>
      <artifactId>spark-core_2.10</artifactId>
      <version>${spark.version}</version>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.spark</groupId>
      <artifactId>spark-mllib_2.10</artifactId>
      <version>${spark.version}</version>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>org.scala-lang</groupId>
      <artifactId>scala-library</artifactId>
      <version>2.10.4</version>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>org.nd4j</groupId>
      <artifactId>nd4j-x86</artifactId>
      <version>${nd4j.version}</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.nd4j</groupId>
      <artifactId>nd4j-api</artifactId>
      <version>${nd4j.version}</version>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>org.deeplearning4j</groupId>
      <artifactId>dl4j-spark-ml</artifactId>
      <version>${dl4j.version}</version>
      <scope>provided</scope>
    </dependency>
  </dependencies>
</project>
`

**Comment by agibsonccc** [comment]
We're going to deploy an update to maven central here tomorrow.
